src/modules/form/components/DynamicField.tsx

```diff
@@ -317,8 +317,11 @@ const DynamicField: React.FC<DynamicFieldProps> = ({
       } else {
         inputComponent = (
           <FormSelect
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-            value={currentValue}
```

Comment:
Eesh yes, there is a bug, but I'm not sure if this is the fix we want.

The call to `getValueFromPickListData` on line 125 is supposed to set the initial value based on the remotely fetched picklist, but it's not being called because the picklist fetch isn't happening anymore since the pick lists are now pre-loaded using `usePreloadPicklists`.

As you can see the `getValueFromPickListData` function has some more complicated logic than simply finding the matching code - it also chooses the initial code, and supports arrays of codes.

A simple fix would be something like updating the usage of `usePickList` hook so that the value gets changed whether the fetch happened or not.

Would you mind digging into this? Otherwise I will create a separate ticket and I can prioritize it for myself this week.